Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/ServiceManifests/ServiceManifestRequiredLegacyConfigurationEntry.cs

```diff
@@ -0,0 +1,25 @@
+﻿// <copyright file="ServiceManifestRequiredLegacyConfigurationEntry.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.TenantManagement.ServiceManifests
+{
+    /// <summary>
+    /// Base class for legacy configuration entry types.
+    /// </summary>
+    public abstract class ServiceManifestRequiredLegacyConfigurationEntry : ServiceManifestRequiredConfigurationEntry
+    {
+        /// <summary>
+        /// The prefix for legacy v2 configuration entries.
+        /// </summary>
+        private const string LegacyV2StorageConfigurationEntryKeyPrefix = "StorageConfiguration__";
+
+        /// <summary>
+        /// Gets the configuration entry key in the style of legacy entries by prefixing it with
+        /// <see cref="LegacyV2StorageConfigurationEntryKeyPrefix"/>.
+        /// </summary>
+        /// <returns>The prefixed key that should be used when adding the configuration to the target tenant.</returns>
+        public string LegacyConfigurationEntryKey =>
+            LegacyV2StorageConfigurationEntryKeyPrefix + this.Key;
```

Comment:
Does this always work?

CosmosDb uses `$"StorageConfiguration__{definition.DatabaseName}__{definition.ContainerName}"`. Table storage uses `$"StorageConfiguration__Table__{definition.TableName}"`.

Wondering if we should drop the 'prefix' idea, and make `LegacyConfigurationEntryKey` abstract, and use the various `TenantPropertyKeyForLogicalContainer` methods defined in the relevant `Corvus.Storage.Azure.*.Tenancy` libraries?

Reply:
As per our offline discussion, we'll add the [StorageType]ContainerDefinition classes (which used to be part of the storage-specific Corvus.Tenancy libraries) so that we have access to the logical container names needed to correctly generate the keys.